ApplicationLogs/LogReader.cs

```diff
@@ -78,7 +78,7 @@ public void OnPersist(Snapshot snapshot, IReadOnlyList<Blockchain.ApplicationExe
                     }
                     return notification;
                 }).ToArray();
-                writeBatch.Put(appExec.Transaction.Hash.ToArray(), json.ToString());
+                writeBatch.Put((appExec.Transaction.Hash ?? snapshot.PersistingBlock.Hash).ToArray(), json.ToString());
```

Comment:
Is not the same expected value

Reply:
Do you have any good ideas?

Reply:
In fact, is stored, but is not readed, but one extra property in the json should be great, something like 
```csharp
json["type"]=appExec.Transaction==null?"tx":"block";
```